pojo-generator-core/src/main/java/uk/gov/justice/generation/pojo/plugin/classmodifying/AddFieldsAndMethodsToClassPlugin.java

```diff
@@ -69,13 +81,30 @@
     private MethodSpec buildConstructor(
             final List<Definition> definitions,
             final ClassNameFactory classNameFactory,
-            final PluginContext pluginContext) {
+            final PluginContext pluginContext,
+            final ClassDefinition classDefinition) {
         final List<String> fieldNames = definitions.stream().map(Definition::getFieldName).collect(toList());
 
         final List<ParameterSpec> constructorParameters = constructorParameters(
                 definitions,
                 classNameFactory,
                 pluginContext);
+
+        if (additionalPropertiesDeterminer.shouldAddAdditionalProperties(classDefinition, pluginContext)) {
+
+            final ParameterizedTypeName map = ParameterizedTypeName.get(
+                    ClassName.get(Map.class),
+                    TypeName.get(String.class),
+                    TypeName.get(Object.class));
+
+            return constructorBuilder()
+                    .addModifiers(PUBLIC)
+                    .addParameters(constructorParameters)
+                    .addParameter(map, ADDITIONAL_PROPERTIES, FINAL)
+                    .addCode(constructorStatements(fieldNames))
+                    .addCode(constructorStatements(Arrays.asList(ADDITIONAL_PROPERTIES)))
```

Comment:
Arrays.asList can be replaced with Collections.singletonList and statically imported